source/common/listener_manager/listener_manager_impl.cc

```diff
@@ -639,9 +630,21 @@ absl::StatusOr<bool> ListenerManagerImpl::addOrUpdateListenerInternal(
   return true;
 }
 
-bool ListenerManagerImpl::hasListenerWithDuplicatedAddress(const ListenerList& list,
+bool ListenerManagerImpl::hasListenerWithDuplicatedAddress(const ListenerList& listener_list,
                                                            const ListenerImpl& listener) {
-  for (const auto& existing_listener : list) {
+  // This is new listener or new version of existing listener but with different addresses
+  // or different socket options.
+  // Check if the listener has duplicated address with existing listeners.
+
+  for (const auto& existing_listener : listener_list) {
+    if (listener.reusePort() && existing_listener->name() == listener.name()) {
```

Comment:
I prefer to reject `reuse_port == false` and the listeners with the same name listen with the duplicated address early as my original PR.

https://github.com/envoyproxy/envoy/blob/50df6826c4861560bce78ace9fb7d807115585e3/source/common/listener_manager/listener_manager_impl.cc#L528-L532

In this implementation, the user can receive only messages that identify "listening address is duplicated". I guess the users may confused about the message, and can't know why this configuration is rejected.
I

Reply:
Oh, But I think this message helps users to understand the root cause. So personally current implementation is LGTM.
https://github.com/envoyproxy/envoy/pull/38601/files#diff-07eac8528f510167dcdc9f00f004e553f18372a90452c7a40af81a91f124c068R1149-R1150